src/components/local-block.tsx

```diff
@@ -110,47 +110,29 @@ export const LocalBlock = (props: LocalBlockProps) => {
   );
 };
 
-const getBlockKey = (block: Block) =>
-  `${block?.owner}/${block?.repo}__${block?.id}`.replace(/\//g, "__");
-const getMetadataPath = (block: Block, path: string) =>
-  `.github/blocks/${block?.type}/${getBlockKey(block)}/${encodeURIComponent(
-    path
-  )}.json`;
 const fetchGitHubData = async (
-  type: string,
-  config: FileContext | FolderContext
+  path: string,
+  params: Record<string, any> = {},
+  id: string = ""
 ) => {
-  if (type === "file-content") {
-    const data = await getFileContent({
-      owner: config.owner,
-      repo: config.repo,
-      path: config.path,
-      fileRef: config.fileRef || "HEAD",
+  try {
+    const apiUrl = `https://api.github.com${path}`;
+
+    const res = await fetch(apiUrl, {
+      ...params,
+      // this callback is limited to GET requests
+      method: "GET",
     });
-    return data;
-  } else if (type === "metadata") {
-    try {
-      const res = await getFileContent({
-        owner: config.owner,
-        repo: config.repo,
-        path: getMetadataPath(config.block, config.path),
-        fileRef: "HEAD",
-        cache: new Date().toString(),
-      });
-      const fullMetadata = JSON.parse((res.content || "{}") as string);
-      return fullMetadata || {};
-    } catch (e) {
-      return {};
-    }
-  } else if (type === "repo-info") {
-    try {
-      const res = await getRepoInfo({
-        owner: config.owner,
-        repo: config.repo,
-      });
-      return res;
-    } catch (e) {
-      return {};
+
+    if (res.status !== 200) {
+      throw new Error(
+        `Error fetching generic GitHub API data: ${apiUrl}\n${await res.text()}`
+      );
```

Comment:
this error gets swallowed by the blanket catch below. (maybe errors should be transmitted back to the calling block? would make it easier to debug blocks that use the API heavily)